projects/core/src/cart/store/actions/cart-voucher.action.ts

```diff
@@ -0,0 +1,72 @@
+import {
+  EntityFailAction,
+  EntityLoadAction,
+  EntitySuccessAction,
+} from '../../../state/utils/entity-loader/entity-loader.action';
+import {
+  LoaderFailAction,
+  LoaderLoadAction,
+  LoaderSuccessAction,
+} from '../../../state/utils/loader/loader.action';
+import { ADD_VOUCHER_PROCESS_ID, CART_DATA } from '../cart-state';
+
+export const ADD_CART_VOUCHER = '[Cart] Add Cart Vouchers';
+export const ADD_CART_VOUCHER_FAIL = '[Cart] Add Cart Voucher Fail';
+export const ADD_CART_VOUCHER_SUCCESS = '[Cart] Add Cart Voucher Success';
+
+export const REMOVE_CART_VOUCHER = '[Cart] Remove Cart Voucher';
+export const REMOVE_CART_VOUCHER_FAIL = '[Cart] Remove Cart Voucher Fail';
+export const REMOVE_CART_VOUCHER_SUCCESS = '[Cart] Remove Cart Voucher Success';
+
+// Adding cart voucher actions
+export class AddCartVoucher extends EntityLoadAction {
+  readonly type = ADD_CART_VOUCHER;
+  constructor(public payload: any) {
```

Comment:
I think “payload:any” is inappropriate

Reply:
Please check and correct all the following "payload"

Reply:
updated